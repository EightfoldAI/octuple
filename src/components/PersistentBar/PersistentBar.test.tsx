import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { PersistentBar, PersistentBarType } from './';
import { render } from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

class ResizeObserver {
  observe() {
    // do nothing
  }
  unobserve() {
    // do nothing
  }
  disconnect() {
    // do nothing
  }
}

window.ResizeObserver = ResizeObserver;

describe('PersistentBar', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Renders without crashing', () => {
    const { container, getByRole } = render(
      <PersistentBar content={'PersistentBar test render'} />
    );
    const infoBar = getByRole('region');
    expect(() => container).not.toThrowError();
    expect(infoBar).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is bottomBarButtonsOnLeft', () => {
    const { container } = render(
      <PersistentBar
        content={'PersistentBar test bottomBarButtonsOnLeft'}
        type={PersistentBarType.bottomBarButtonsOnLeft}
      />
    );
    expect(container.querySelector('.bottom-left-align')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is bottomBarButtonsOnRight', () => {
    const { container } = render(
      <PersistentBar
        content={'PersistentBar test bottomBarButtonsOnRight'}
        type={PersistentBarType.bottomBarButtonsOnRight}
      />
    );
    expect(container.querySelector('.bottom-right-align')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is bottomBarSecondaryButtons', () => {
    const { container } = render(
      <PersistentBar
        content={'PersistentBar test bottomBarSecondaryButtons'}
        type={PersistentBarType.bottomBarSecondaryButtons}
      />
    );
    expect(container.querySelector('.bottom-secondary-buttons')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is bottomBarWithText', () => {
    const { container } = render(
      <PersistentBar
        content={'PersistentBar test bottomBarWithText'}
        type={PersistentBarType.bottomBarWithText}
      />
    );
    expect(container.querySelector('.bottom-with-text')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is topBarButtons', () => {
    const { container } = render(
      <PersistentBar
        content={'PersistentBar test topBarButtons'}
        type={PersistentBarType.topBarButtons}
      />
    );
    expect(container.querySelector('.top-button-menu')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is topBarPagination', () => {
    const { container } = render(
      <PersistentBar
        content={'PersistentBar test topBarPagination'}
        type={PersistentBarType.topBarPagination}
      />
    );
    expect(container.querySelector('.top-pagination')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is topBarWithText', () => {
    const { container } = render(
      <PersistentBar
        content={'PersistentBar test topBarWithText'}
        type={PersistentBarType.topBarWithText}
      />
    );
    expect(container.querySelector('.top-with-text')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('PersistentBar is custom', () => {
    const { container } = render(
      <PersistentBar
        children={<div className="custom">PersistentBar test custom</div>}
      />
    );
    expect(container.querySelector('.custom')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
