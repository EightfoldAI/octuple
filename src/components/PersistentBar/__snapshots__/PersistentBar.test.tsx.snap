// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersistentBar PersistentBar is bottomBarButtonsOnLeft 1`] = `
<div>
  <div
    class="persistent-bar bottom-left-align"
    role="region"
  />
</div>
`;

exports[`PersistentBar PersistentBar is bottomBarButtonsOnRight 1`] = `
<div>
  <div
    class="persistent-bar bottom-right-align"
    role="region"
  />
</div>
`;

exports[`PersistentBar PersistentBar is bottomBarSecondaryButtons 1`] = `
<div>
  <div
    class="persistent-bar bottom-secondary-buttons"
    role="region"
  />
</div>
`;

exports[`PersistentBar PersistentBar is bottomBarWithText 1`] = `
<div>
  <div
    class="persistent-bar bottom-with-text"
    role="region"
  >
    <div>
      <h4 />
      <div
        class="content"
      >
        PersistentBar test bottomBarWithText
      </div>
    </div>
    <div />
  </div>
</div>
`;

exports[`PersistentBar PersistentBar is custom 1`] = `
<div>
  <div
    class="persistent-bar bottom-with-text"
    role="region"
  >
    <div
      class="custom"
    >
      PersistentBar test custom
    </div>
  </div>
</div>
`;

exports[`PersistentBar PersistentBar is topBarButtons 1`] = `
<div>
  <div
    class="persistent-bar top-button-menu"
    role="region"
  />
</div>
`;

exports[`PersistentBar PersistentBar is topBarPagination 1`] = `
<div>
  <div
    class="persistent-bar top-pagination"
    role="region"
  >
    <div
      class="pagination"
      role="navigation"
    >
      <span />
      <ul
        class="pager"
      >
        <li>
          <button
            aria-current="true"
            aria-disabled="false"
            class="pagination-button active button button-neutral button-medium"
          >
            <span
              class="button-text-medium"
            >
              1
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`PersistentBar PersistentBar is topBarWithText 1`] = `
<div>
  <div
    class="persistent-bar top-with-text"
    role="region"
  >
    <div
      class="stack full-width horizontal undefined gap-m"
      style="justify-content: space-between;"
    >
      <button
        aria-disabled="false"
        class="icon-button button button-system-ui button-medium round-shape icon-left"
      >
        <span
          aria-hidden="false"
          class="icon icon-wrapper"
          role="presentation"
        >
          <svg
            role="presentation"
            style="width: 20px; height: 20px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
              style="fill: currentColor;"
            />
          </svg>
        </span>
      </button>
      <div>
        <h4 />
        <div
          class="content"
        >
          PersistentBar test topBarWithText
        </div>
      </div>
      <div
        class="pagination"
        role="navigation"
      >
        <span />
        <ul
          class="pager page-tracker"
        >
          <li>
            <span>
              1
            </span>
             
            <span>
              of
            </span>
             
            <span>
              1
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`PersistentBar Renders without crashing 1`] = `
<div>
  <div
    class="persistent-bar bottom-with-text"
    role="region"
  >
    <div>
      <h4 />
      <div
        class="content"
      >
        PersistentBar test render
      </div>
    </div>
    <div />
  </div>
</div>
`;
